reverse_replication/launcher.go

```diff
@@ -0,0 +1,273 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	dataflow "cloud.google.com/go/dataflow/apiv1beta3"
+	"cloud.google.com/go/dataflow/apiv1beta3/dataflowpb"
+	"cloud.google.com/go/pubsub"
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	"cloud.google.com/go/storage"
+	adminpb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+var (
+	projectId            string
+	dataflowRegion       string
+	jobNamePrefix        string
+	changeStreamName     string
+	instanceId           string
+	dbName               string
+	metadataInstance     string
+	metadataDatabase     string
+	pubSubDataTopicId    string
+	pubSubEndpoint       string
+	sourceShardsFilePath string
+	sessionFilePath      string
+)
+
+func setupGlobalFlags() {
+	flag.StringVar(&projectId, "projectId", "", "projectId")
+	flag.StringVar(&dataflowRegion, "dataflowRegion", "", "region for dataflow jobs")
+	flag.StringVar(&jobNamePrefix, "jobNamePrefix", "reverse-rep", "job name prefix for the dataflow jobs")
+	flag.StringVar(&changeStreamName, "changeStreamName", "", "change stream name")
+	flag.StringVar(&instanceId, "instanceId", "", "spanner instance id")
+	flag.StringVar(&dbName, "dbName", "", "spanner database name")
+	flag.StringVar(&metadataInstance, "metadataInstance", "", "spanner instance name to store changestream metadata")
+	flag.StringVar(&metadataDatabase, "metadataDatabase", "", "spanner database name to store changestream metadata")
+	flag.StringVar(&pubSubDataTopicId, "pubSubDataTopicId", "", "pub/sub data topic id, should be of the form projects/my-project/topics/my-topic")
+	flag.StringVar(&pubSubEndpoint, "pubSubEndpoint", "", "pub/sub endpoint, defaults to same endpoint as the dataflow region.")
+	flag.StringVar(&sourceShardsFilePath, "sourceShardsFilePath", "", "gcs file path for file containing shard info")
+	flag.StringVar(&sessionFilePath, "sessionFilePath", "", "gcs file path for session file generated via HarbourBridge")
+}
+
+func prechecks() error {
+	if projectId == "" {
+		return fmt.Errorf("please specify a valid projectId")
+	}
+	if dataflowRegion == "" {
+		return fmt.Errorf("please specify a valid dataflowRegion")
+	}
+	if changeStreamName == "" {
+		return fmt.Errorf("please specify a valid changeStreamName")
+	}
+	if instanceId == "" {
+		return fmt.Errorf("please specify a valid instanceId")
+	}
+	if dbName == "" {
+		return fmt.Errorf("please specify a valid dbName")
+	}
+	if metadataInstance == "" {
+		return fmt.Errorf("please specify a valid metadataInstance")
+	}
+	if metadataDatabase == "" {
+		return fmt.Errorf("please specify a valid metadataDatabase")
+	}
+	if pubSubDataTopicId == "" {
+		return fmt.Errorf("please specify a valid pubSubDataTopicId of the form project/project-name/topics/topic-name")
```

Comment:
Can we add this description of the topic name format in the Readme and flags definition